doc/proposals/cli-ux-phase1.md

```diff
@@ -0,0 +1,116 @@
+
+#### Flags for `olm init`
+
+| Flag             | Type   | Description                                                                                     |
+|------------------|--------|-------------------------------------------------------------------------------------------------|
+| `--cluster-type` | string | The type of cluster to deploy OLM to, one of `upstream`, `ocp`, `okd` (default `upstream`)      |
+| `--olm-version`  | string | The version of OLM to deploy (default `latest`)                                                 |
+| `--kubeconfig`   | string | Path for custom Kubernetes client config file (overrides the default locations)                 |
+
+### `operator-sdk alpha olm up`
+
+Creates all necessary resources in the cluster to run the operator via OLM, waits for the operator to be deployed by OLM, and tails the operator log, similar to `operator-sdk up local`. When the user terminates the process or if a timeout or error occurs, all of the created resources will be cleaned up.
+
+#### Prerequisites
+
+1. An operator container image built and pushed to a registry accessible to the cluster.
+2. An operator bundle on disk (created by `operator-sdk olm-catalog gen-csv`).
+
+#### Flags for `olm up`
+
+| Flag           | Type   | Description                                                                                     |
+|----------------|--------|-------------------------------------------------------------------------------------------------|
+| `--bundle-dir` | string | The directory containing the operator bundle (default `./deploy/olm-catalog/<operator-name>`)   |
+| `--kubeconfig` | string | Path for custom Kubernetes client config file (overrides the default locations)                 |
+| `--namespace`  | string | Namespace in which to deploy operator and RBAC rules (overrides namespace from current context) |
+
+#### Resources
+
+OLM uses Kubernetes APIs to learn about the set of operators that are available to be installed and to manage operator lifecycles (i.e. install, upgrade, uninstall). The following resources will be created in the cluster.
+
+| API Kind        | Description  |
+|-----------------|--------------|
+| `ConfigMap`     | Contains catalog data created from the on-disk operator bundle. |
+| `CatalogSource` | Tells OLM where to find operator catalog data. This will refer to the catalog `ConfigMap`. |
+| `OperatorGroup` | Tells OLM which namespaces the operator will have RBAC permissions for. We will set this up with the namespace of the current context from the user's `$KUBECONFIG`. |
+| `Subscription`  | Tells OLM to manage installation and upgrade of an operator in the namespace in which the `Subscription` is created. We'll create it in the namespace of the current context from the user's `$KUBECONFIG`. |
+
+**Open question:** Should we handle cleanup for any of the InstallPlan, CSV, CRD, and CR resources? Which of these will be automatically garbage-collected?
```

Comment:
Cleanup when the command is aborted by the user?

Reply:
Yes, I'll update to clarify.